### DISCUSSION TITLE
Arbitrary JSON

### DISCUSSION THREAD

**Discussion by Minivera** [encoredev/encore#26] [Help] [answered]
Hi,

I noticed that encore does not support arbitrary JSON in schemas for both responses or parameters. That can be very useful when building APIs that consume arbitrary schemas or things like annotations. In Go, I would usually try to support those with a `interface{}` type, but encore triggers a `cannot use interface types in Encore schema definitions` error.

Is support for arbitrary JSON structure currently planned for encore? If you, would you welcome PRs to implement this feature?

Thanks

**Comment by eandre** [comment, marked as answer]
You can use `json.RawMessage` to communicate that a particular field holds raw, arbitrary JSON. And then you can parse it yourself in the handler in whatever way is best for you.

Hope that helps!

**Reply by Minivera** [reply]
Oh! I will try that, thanks!

I'll write back if I run into any issues. I might try to create a docs page for this if they're open source
